### PR TITLE
dm/test: swap upstream and downstream of sync_diff in some cases

### PR DESCRIPTION
**Pull request by lance6716**
### What problem does this PR solve? 
close https://github.com/pingcap/ticdc/issues/3335
close https://github.com/pingcap/ticdc/issues/3409
close https://github.com/pingcap/ticdc/issues/3394

also close https://github.com/pingcap/ticdc/pull/3405
### What is changed and how it works?
as title

### Check List 

Tests 

 - No code

Code changes

Side effects

Related changes

 - Need to cherry-pick to the release branch

### Release note 

```release-note
`None`.
```

### CODE REVIEW

**Comment by GMHDBJD** [comment]
dm_syncer
```["failed to initialize diff process"] [error="no table need to be compared"] [errorVerbose="no table need to be compared\ngithub.com/pingcap/tidb-tools/sync_diff_inspector/source.NewSources\n\t/home/jenkins/agent/workspace/optimization-build-tidb-linux-amd/go/src/github.com/pingcap/tidb-tools/sync_diff_inspector/source/source.go:150\nmain.(*Diff).init\n\t/home/jenkins/agent/workspace/optimization-build-tidb-linux-amd/go/src/github.com/pingcap/tidb-tools/sync_diff_inspector/diff.go:139\nmain.NewDiff\n\t/home/jenkins/agent/workspace/optimization-build-tidb-linux-amd/go/src/github.com/pingcap/tidb-tools/sync_diff_inspector/diff.go:97\nmain.checkSyncState\n\t/home/jenkins/agent/workspace/optimization-build-tidb-linux-amd/go/src/github.com/pingcap/tidb-tools/sync_diff_inspector/main.go:92\nmain.main\n\t/home/jenkins/agent/workspace/optimization-build-tidb-linux-amd/go/src/github.com/pingcap/tidb-tools/sync_diff_inspector/main.go:79\nruntime.main\n\t/usr/local/go/src/runtime/proc.go:225\nruntime.goexit\n\t/usr/local/go/src/runtime/asm_amd64.s:1371"] [stack="main.checkSyncState\n\t/home/jenkins/agent/workspace/optimization-build-tidb-linux-amd/go/src/github.com/pingcap/tidb-tools/sync_diff_inspector/main.go:95\nmain.main\n\t/home/jenkins/agent/workspace/optimization-build-tidb-linux-amd/go/src/github.com/pingcap/tidb-tools/sync_diff_inspector/main.go:79\nruntime.main\n\t/usr/local/go/src/runtime/proc.go:225"]```

File: dm/tests/full_mode/run.sh [hunk 1]
```diff
@@ -141,6 +143,7 @@ function empty_data() {
 	check_log_contains $WORK_DIR/worker1/log/dm-worker.log "progress=\"100.00 %\""
 	check_log_contains $WORK_DIR/worker2/log/dm-worker.log "progress=\"100.00 %\""
 
+	cleanup_data full_mode
```
Comment (lichunzhu) [inline dm/tests/full_mode/run.sh:146]: why do this twice?

**Review by lichunzhu — APPROVED** [review]

File: dm/tests/full_mode/run.sh [hunk 2]
```diff
@@ -141,6 +143,7 @@ function empty_data() {
...
 	cleanup_data full_mode
```
Comment (lance6716) [inline dm/tests/full_mode/run.sh:147]:
```suggestion
	cleanup_data_upstream full_mode
```

File: dm/tests/adjust_gtid/run.sh [hunk 3]
```diff
@@ -84,10 +84,13 @@ function run() {
 	# avoid cannot unmarshal !!str `binlog-...` into uint32 error
 	sed -i "s/binlog-pos-placeholder-1/4/g" $WORK_DIR/dm-task.yaml
 	sed -i "s/binlog-pos-placeholder-2/4/g" $WORK_DIR/dm-task.yaml
-	dmctl_start_task "$WORK_DIR/dm-task.yaml" "--remove-meta"
+	# start DM task. don't check error because it will meet injected error soon
+	run_dm_ctl $WORK_DIR "127.0.0.1:$MASTER_PORT" \
+		"start-task $WORK_DIR/dm-task.yaml --remove-meta"
 
 	# use sync_diff_inspector to check full dump loader
-	check_sync_diff $WORK_DIR $cur/conf/diff_config.toml
```
Comment (Ehco1996) [inline dm/tests/adjust_gtid/run.sh:90]: seems we don't need  `diff_config.toml`, can we delete that file?

File: dm/tests/dm_syncer/run.sh [hunk 4]
```diff
@@ -71,7 +71,7 @@ function run() {
 	run_dm_syncer $WORK_DIR/syncer2 $WORK_DIR/dm-syncer-2.toml $meta_file --syncer-config-format syncer2
 
 	# wait for dm_syncer to init and start
-	sleep 5
+	sleep 10
```
Comment (Ehco1996) [inline dm/tests/dm_syncer/run.sh:74]: in our doc: `Syncer is no longer maintained. Its features are completely superseded by TiDB Data Migration. It is strongly recommended that you use TiDB Data Migration instead`

just a question do we have plan to delete the code releated to `dm-sycner` from our codebase and `syncer-bin` from `tidb-enterprise-tools`?

also cc @sunzhaoyang

File: dm/tests/full_mode/run.sh [hunk 5]
```diff
@@ -104,7 +105,6 @@ function escape_schema() {
 
 	# start DM task only
 	dmctl_start_task "$WORK_DIR/dm-task.yaml" "--remove-meta"
-	check_sync_diff $WORK_DIR $WORK_DIR/diff_config.toml
```
Comment (Ehco1996) [inline dm/tests/full_mode/run.sh:107]: ditto and in incremental_mode

**Review by Ehco1996 — COMMENTED** [review]
restLGTM

File: dm/tests/adjust_gtid/run.sh (hunk 3 above)
Comment (lance6716) [reply dm/tests/adjust_gtid/run.sh:90]: we'll use it after sync_diff fix the behaviour soon

**Review by Ehco1996 — APPROVED** [review]

**Comment by Ehco1996** [comment]
```
[2021-11-12T02:43:45.884Z] Exception:

[2021-11-12T02:43:45.884Z] Traceback (most recent call last):

[2021-11-12T02:43:45.884Z]   File "/usr/lib/python2.7/site-packages/pip/basecommand.py", line 215, in main

[2021-11-12T02:43:45.884Z]     status = self.run(options, args)

[2021-11-12T02:43:45.884Z]   File "/usr/lib/python2.7/site-packages/pip/commands/install.py", line 307, in run

[2021-11-12T02:43:45.884Z]     requirement_set.prepare_files(finder)

[2021-11-12T02:43:45.884Z]   File "/usr/lib/python2.7/site-packages/pip/req/req_set.py", line 370, in prepare_files

[2021-11-12T02:43:45.884Z]     ignore_dependencies=self.ignore_dependencies))

[2021-11-12T02:43:45.884Z]   File "/usr/lib/python2.7/site-packages/pip/req/req_set.py", line 587, in _prepare_file

[2021-11-12T02:43:45.884Z]     session=self.session, hashes=hashes)

[2021-11-12T02:43:45.884Z]   File "/usr/lib/python2.7/site-packages/pip/download.py", line 811, in unpack_url

[2021-11-12T02:43:45.884Z]     hashes=hashes

[2021-11-12T02:43:45.884Z]   File "/usr/lib/python2.7/site-packages/pip/download.py", line 650, in unpack_http_url

[2021-11-12T02:43:45.884Z]     hashes)

[2021-11-12T02:43:45.884Z]   File "/usr/lib/python2.7/site-packages/pip/download.py", line 892, in _download_http_url

[2021-11-12T02:43:45.884Z]     _download_url(resp, link, content_file, hashes)

[2021-11-12T02:43:45.884Z]   File "/usr/lib/python2.7/site-packages/pip/download.py", line 596, in _download_url

[2021-11-12T02:43:45.884Z]     hashes.check_against_chunks(downloaded_chunks)

[2021-11-12T02:43:45.884Z]   File "/usr/lib/python2.7/site-packages/pip/utils/hashes.py", line 49, in check_against_chunks

[2021-11-12T02:43:45.884Z]     for chunk in chunks:

[2021-11-12T02:43:45.885Z]   File "/usr/lib/python2.7/site-packages/pip/download.py", line 564, in written_chunks

[2021-11-12T02:43:45.885Z]     for chunk in chunks:

[2021-11-12T02:43:45.885Z]   File "/usr/lib/python2.7/site-packages/pip/download.py", line 553, in resp_read

[2021-11-12T02:43:45.885Z]     decode_content=False):

[2021-11-12T02:43:45.885Z]   File "/usr/lib/python2.7/site-packages/pip/_vendor/requests/packages/urllib3/response.py", line 353, in stream

[2021-11-12T02:43:45.885Z]     data = self.read(amt=amt, decode_content=decode_content)

[2021-11-12T02:43:45.885Z]   File "/usr/lib/python2.7/site-packages/pip/_vendor/requests/packages/urllib3/response.py", line 320, in read

[2021-11-12T02:43:45.885Z]     flush_decoder = True

[2021-11-12T02:43:45.885Z]   File "/usr/lib64/python2.7/contextlib.py", line 35, in __exit__

[2021-11-12T02:43:45.885Z]     self.gen.throw(type, value, traceback)

[2021-11-12T02:43:45.885Z]   File "/usr/lib/python2.7/site-packages/pip/_vendor/requests/packages/urllib3/response.py", line 242, in _error_catcher

[2021-11-12T02:43:45.885Z]     raise ReadTimeoutError(self._pool, None, 'Read timed out.')

[2021-11-12T02:43:45.885Z] ReadTimeoutError: HTTPSConnectionPool(host='files.pythonhosted.org', port=443): Read timed out.
```

i will add retry for this

**Comment by lance6716** [comment]
/run-integration-test

upload coverage error...

**Pull request merged**